libs/linearSolver/okl/linearSolverUpdateMINRES.okl

```diff
@@ -0,0 +1,80 @@
+/*
+
+The MIT License (MIT)
+
+Copyright (c) 2017 Tim Warburton, Noel Chalmers, Jesse Chan, Ali Karakus
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+
+*/
+
+#define p_NT 256
+
+@kernel void updateMINRES(const dlong Ntotal,
+                          const dfloat ma2,   // -a2
+                          const dfloat ma3,   // -a3
+                          const dfloat alpha, // -del/gam
+                          const dfloat beta,  // -gam/gamp
+                          @restrict dfloat *z,
+                          @restrict dfloat *q_old,
+                          @restrict dfloat *q,
+                          @restrict dfloat *r_old,
+                          @restrict dfloat *r,
+                          @restrict dfloat *p)
+{
+  for (int no = 0; no < Ntotal; no += p_NT; @outer(0)) {
+    for (int n = no; n < no + p_NT; n++; @inner(0)) {
+      if (n < Ntotal) {
```

Comment:
These three lines can actually be compressed to
```suggestion
  for(dlong n=0;n<Ntotal;++n;@tile(p_NT,@outer,@inner)){
```
OCCA will add the bounds check for you.

Reply:
Nice.  I'd seen the `@tile` feature before but did not know that it would do the bounds check for you---that's handy.  I'll change this shortly.